### ISSUE TITLE
Whether I can provide complete train code (keras version) or not, I always input the wrong dimension.entreat

### ISSUE THREAD

**Issue by duxingqi517** [GeoZcx/A-deeply-supervised-image-fusion-network-for-change-detection-in-remote-sensing-images#12]
I superimposed the two images into an array and imported the model. Always report errors as follows:

Input 0 of layer block1_pool is incompatible with the layer: expected ndim=4, found ndim=5. Full shape received: [4, 2, 512, 512, 64]

This is my code: model.fit(train_imag,train_label,batch_size = 4,epochs=100)

`def traindataloard(path1, path2):      
    img1path = os.listdir(path1)      
    img2path = os.listdir(path2)
    img1group = [input_t1 + name1 for name1 in img1path]        
    img2group = [input_t2 + name2 for name2 in img2path]
    train_img2 = []
    for img1n, img2n in zip(img1group, img2group):    
        train_img1 = []
        img1 = cv2.imread(img1n)
        img2 = cv2.imread(img2n)

        img1 = np.array(img1)  
        img2 = np.array(img2)

        train_img = [img1, img2]    
        train_img2.append(train_img)
    return train_img2`

train_img = traindataloard(input_t1, input_t2)

**Comment by GeoZcx** [comment]
I suggest you use generators to yield a batch of training samples for each epoch. One generator for training data loading, and the other one for training label loading.
